packages/context/src/resolution-session.ts

```diff
@@ -342,4 +342,9 @@ export interface ResolutionOptions {
    * will return `undefined` instead of throwing an error.
    */
   optional?: boolean;
+
+  /**
+   * Environment for resolution, such as `dev`, `test`, `staging`, and `prod`
+   */
+  environment?: string;
```

Comment:
I am not happy about introducing `environment` to LoopBack runtime. From what I know, this approach is considered as an anti-pattern. It couples internal implementation of our framework with the environment-based decision making. Also in many cases the same environment may need different configuration, e.g. the application can be deployed in `production` environment to different geographic zones and thus will need different database connection strings.

IMO, all environment-specific aspects should be handled in a single place, and that's the single place where we are loading the configuration - e.g. from env variables, JSON files, Consul service, whatever. The concept of per-environment config should stay inside that layer and should not leak to our IoC implementation.

This is another reason why I want to address the "configurability" feature from a high-level perspective of users first (see my comment above).

Reply:
I removed the `env`.